habitat-lab/habitat/tasks/rearrange/rearrange_sensors.py

```diff
@@ -247,10 +247,12 @@ def _get_observation_space(self, *args, config, **kwargs):
         )
 
     def get_observation(self, observations, episode, *args, **kwargs):
-        joints_pos = self._sim.get_agent_data(
-            self.agent_id
-        ).articulated_agent.get_joint_transform()[0]
-        return np.array(joints_pos, dtype=np.float32)
+        curr_agent = self._sim.get_agent_data(self.agent_id).articulated_agent
+        if hasattr(curr_agent, "get_joint_transform"):
```

Comment:
Can you check for the type with `is instance` instead of `hasattr`? This will be more clear what interface you expect here.